File: banchus-app/src/main/java/pe/nanamochi/banchus/service/PacketBundleService.java
package pe.nanamochi.banchus.service;

import java.util.List;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import pe.nanamochi.banchus.redis.model.PacketBundle;
import pe.nanamochi.banchus.redis.repository.PacketBundleRepository;

@Service
@RequiredArgsConstructor
public class PacketBundleService {
  private final PacketBundleRepository packetBundleRepository;

  public void enqueue(UUID sessionId, PacketBundle packetBundle) {
    packetBundleRepository.enqueue(sessionId, packetBundle);
  }

  public List<PacketBundle> dequeueAll(UUID sessionId) {
    return packetBundleRepository.dequeueAll(sessionId);
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/service/PacketBundleService.kt
package pe.nanamochi.banchus.service

import java.util.UUID
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import pe.nanamochi.banchus.redis.entity.PacketBundle
import pe.nanamochi.banchus.redis.repository.PacketBundleRepository

@Service
class PacketBundleService(private val packetBundleRepository: PacketBundleRepository) {
    private val log = LoggerFactory.getLogger(javaClass)

    fun enqueue(sessionId: UUID, packetBundle: PacketBundle) =
        packetBundleRepository.enqueue(sessionId, packetBundle)

    fun dequeueAll(sessionId: UUID): List<PacketBundle> =
        packetBundleRepository.dequeueAll(sessionId)
}
